drivers/ra/fsp/src/r_ether_phy/r_ether_phy.c

```diff
@@ -0,0 +1,1112 @@
+/*
```

Comment:
Including this file in a hal module is not inhibit, but using it would be unpreferable since the phy interface already exists.

Reply:
Hello @soburi, I am sorry for late responding. I saw that some vendors (ex. nxp and microchip) do similar:
![image](https://github.com/user-attachments/assets/b80f0d7d-8c52-4b17-9df9-520c64efdd3e)

![image](https://github.com/user-attachments/assets/a97ae543-a844-4452-8e3d-11f06c609c7a)

In additional, we didn't find target Phy match with our board (ICS1894) in Zephyr repos. So re-using FSP is best choice for us. Please correct me if I had misunderstanding with you.

Reply:
> Hello @soburi, I am sorry for late responding. I saw that some vendors (ex. nxp and microchip) do similar:

As I wrote, it's not a problem if the file is included here.

The driver for ksz8081 is https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ethernet/phy/phy_microchip_ksz8081.c, and in fact it is referenced here:

https://github.com/zephyrproject-rtos/zephyr/blob/bd03883744d49598dbb6755632e25469159906cc/boards/nxp/mimxrt1020_evk/mimxrt1020_evk.dts#L148

The file you show probably didn't compile.

I haven't fully investigated the microchip version, but I think it's probably the same.

> In additional, we didn't find target Phy match with our board (ICS1894) in Zephyr repos. So re-using FSP is best choice for us. Please correct me if I had misunderstanding with you.

So, My understanding is "You also need to create this phy driver for zephyr if you want support the ethernet mac.".


But I'm not maintainer of ethernet, I can't make decision.
so I suggest to consult to the ethenet maintainer.

Reply:
Thank you so much for your comment. I understood. In this case, I will enable review in Zephyr PR and get more comments from ethernet maintainers.